### ISSUE TITLE
[regression between 4.6.0 and 4.9.0] Build breaks on FreeBSD/arm64: modules/core/src/system.cpp:573:13: error: unknown type

### ISSUE THREAD

**Issue by yurivict** [opencv/opencv#25527]
### System Information

OpenCV version: 4.9.0
FreeBSD 15

### Detailed description

 ```
/construction/xports/graphics/opencv/work/opencv-4.9.0/modules/core/src/system.cpp:573:13: error: unknown type 
name 'Elf64_auxv_t'
            Elf64_auxv_t auxv;
```
The ```#include <gelf.h>``` statement for ```Elf64_auxv_t``` is missing.
The search for the package ```elfutils``` which installs the header ```gelf.h``` is also missing.

The code section beginning with ```int cpufile = open("/proc/self/auxv", O_RDONLY);``` is new in 4.9.0

Additionally, when the file /proc/self/auxv doesn't exist on FreeBSD, it should print that ```linprocfs``` should be enabled, because /proc/self/auxv is only created when ```linprocfs``` is enabled. The current code is silent in case of the failure to open this file.

In short, 3 items need to be fixed:
1) Add the include for ```Elf64_auxv_t```
2) Add search for the ```elfutils``` in configure
3) Add the user warning when "/proc/self/auxv" isn't available on FreeBSD suggesting to enable ```linprocfs```

### Steps to reproduce

n/a

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [X] There is reproducer code and related data files (videos, images, onnx, etc)

**Comment by asmorkalov** [comment]
Unfortunately BSD systems are out for OpenCV team CI Scope. Feel free to contribute solution.

**Comment by asmorkalov** [comment]
Fixed in https://github.com/opencv/opencv/pull/24819
